### PR TITLE
Break dependency on lit/localize

### PR DESCRIPTION
**Pull request by oyamauchi**
## Description

This is unfortunately a huge PR, so here's a detailed guide.

Using `@lit/localize` at runtime also entails depending on the
entirety of Lit; the localize package is not factored so that you can
just use the runtime string-substitution part. So we gotta get rid of it.

The idea here was to be minimally disruptive by simply swapping out
the `msg` and `str` functions from lit-localize with our own
implementations.  That's in `src/i18n/use-translated.ts` and
`src/i18n/str.ts`. The `strings:build` command now also invokes sed to
adjust the import in the generated translated-strings files.

Rather than putting the current locale in global state like Lit does,
I put it in React context, accessed via the new custom hook
`useTranslated`.  The hook returns the `msg` function, bound to the
current locale; callsites then use that msg function as before.

Most of the changed files are just plumbing in the new hook and/or msg
function. The most substantive changes are in:

- `state-calculator.tsx` where the locale context is set

- `calculator.tsx` -- this is the old embed, which was never actually
  localized; I removed all the `msg` calls here since we're getting
  rid of it Soon™

- `i18n/use-translated.ts` -- the core of the new stuff

## Test Plan

- Run `yarn strings:extract` and make sure there are no changes to es.xlf.

- Run `yarn strings:build` and make sure there are no changes.

- View the calculator and hit Calculate; make sure all strings come up
  looking correct in English. Add the `lang="es"` attribute to the
  calculator element and make sure the UI immediately switches to
  Spanish, except for incentive blurbs (which come from the API).

  Hit Calculate again and make sure the incentive blurbs are now in Spanish.

### CODE REVIEW

File: src/state-incentive-details.tsx [hunk 1]
```diff
@@ -187,28 +182,37 @@ const LinkButton: FC<PropsWithChildren<{ href: string }>> = ({
   </a>
 );
 
-const renderIncentiveCard = (key: Key, incentive: Incentive) => (
```
Comment (oyamauchi) [inline src/state-incentive-details.tsx:190]: I turned this into its own component because passing the Key around felt like not what you're supposed to do

File: src/state-incentive-details.tsx [hunk 2]
```diff
@@ -301,7 +306,9 @@ const renderCardCollection = (incentives: Incentive[]) => (
     {incentives
       // Put IRA rebates after everything else
       .sort((a, b) => +isIRARebate(a) - +isIRARebate(b))
-      .map((incentive, index) => renderIncentiveCard(index, incentive))}
+      .map((incentive, index) => (
+        <IncentiveCard key={index} incentive={incentive} />
```
Comment (oyamauchi) [inline src/state-incentive-details.tsx:310]: I know you're not supposed to use a numerical index as a key, but incentives don't have identifiers for now

File: src/state-calculator.tsx [hunk 3]
```diff
@@ -397,7 +393,6 @@ export const StateCalculator: FC<{
           stateId={stateId}
           initialValues={getInitialFormValues()}
           showEmailField={!!showEmail && !emailSubmitted}
-          showProjectField={true}
```
Comment (oyamauchi) [inline src/state-calculator.tsx:400]: Got rid of this as a drive-by change since it's always true now

File: src/icon-tab-bar.tsx [hunk 4]
```diff
@@ -42,23 +43,23 @@ export const IconTabBar: FC<Props> = ({ tabs, selectedTab, onTabSelected }) => {
         )}
         role="tab"
         aria-selected={isSelected}
-        aria-label={PROJECTS[project].label()}
+        aria-label={PROJECTS[project].label(msg)}
```
Comment (RandomEtc) [inline src/icon-tab-bar.tsx:46]: Reading these as "label using msg" - would it be clearer as `msg(PROJECTS[project].label)` to avoid passing the `msg` helper into every label?

Comment (RandomEtc) [reply src/icon-tab-bar.tsx:46]: Oh I see, because we're still using lit localize at build time to extract strings, we still need a `msg` call site to allow it to know which strings need translations. Is that it?

File: src/state-incentive-details.tsx (hunk 2 above)
Comment (RandomEtc) [reply src/state-incentive-details.tsx:310]: I'd be in favor of returning one!

File: package.json [hunk 5]
```diff
@@ -24,7 +23,7 @@
     "lint": "tsc --noEmit && prettier --check . && eslint .",
     "prepare": "husky install",
     "strings:extract": "lit-localize extract",
-    "strings:build": "lit-localize build && prettier --write src/locales"
+    "strings:build": "lit-localize build && sed -e 's!@lit/localize!../str!' -i '' src/i18n/strings/*.ts && prettier --write src/i18n"
```
Comment (RandomEtc) [inline package.json:26]: Do we get this from a dev dependency now?

**Review by RandomEtc — APPROVED** [review]
I think this makes sense. Definitely jumping through a few hoops to use the `msg` function to tag things. I wonder if there's a way to use tagged strings to make them magically re-render based on context? Or if we should evaluate an alternative approach - I've used `t('some.string.id')` in other apps, with every language in a strings file (yaml or json) and english doesn't need to be hard-coded at the point of use. That might help avoid threading the msg function through, if we want to part ways with the lit toolchain entirely?

But this seems fine to keep us moving toward React-all-the-things.

**Comment by RandomEtc** [comment]
Note: I'm relying on the Cypress test here to verify the translated behavior is actually working, since we don't have a preview link with Spanish. It might be worth putting the preview build into a local copy of our consumer site to test more thoroughly?

File: package.json (hunk 5 above)
Comment (oyamauchi) [reply package.json:26]: You mean `sed`? It's included in every Unix-like OS. (Also this command is only run in local dev, not in CI, so it only needs to work on macOS.)

To be clear, the command is making this change: [src/i18n/strings/es.ts](https://github.com/rewiringamerica/embed.rewiringamerica.org/pull/126/files#diff-056f9702ed61bdcb40e260d5b118d5971d61a4088e7e00c4479b92c25e870707)

File: src/icon-tab-bar.tsx (hunk 4 above)
Comment (oyamauchi) [reply src/icon-tab-bar.tsx:46]: Right, we're still using `lit/localize-tools`, which needs the argument to `msg` to be a string literal so it can extract it.

**Review by oyamauchi — COMMENTED** [review]
I admit this is a bit of a weird Frankenstein approach, but my rationale was:

- This minimized the grunt work of going around changing every single `msg` callsite
- It lets us keep using the same build-time translation workflow/tooling
- I looked at alternatives, including the apparent standard `react-i18next`. It uses the `t('string.id')` approach, which I actually _don't_ like -- I think having the user-visible English string in situ is a plus. You can see what string will show up just by scanning the usage site; you don't need to flip to another file.

This approach has the key limitation that it doesn't handle localizing strings that need to interpolate HTML, rather than just a string. (We had this problem in a recent PR with needing to localize `View our <a href="...">terms</a>`.)

We may very well switch to another localization framework at some point, but I think this is the quickest reasonable way forward for now.

File: package.json (hunk 5 above)
Comment (RandomEtc) [reply package.json:26]: No I meant the `lit-localize` command - I see there's still `lit/localize-tools` in dev deps, which is what I was checking. From the diff I could only see the removal of the runtime package.
